### DISCUSSION TITLE
how to open capture window using lua function?

### DISCUSSION THREAD

**Discussion by rogtino** [nvim-orgmode/orgmode#761] [Q&A] [answered]
Is there an api or some ways to open capture window using lua function?

**Comment by chipsenkbeil** [comment]
You want to create a new `OrgCapture` instance, populate it with files and templates, and then execute `capture:prompt()`. You can find an example from how I use it in org roam below:

https://github.com/chipsenkbeil/org-roam.nvim/blob/45153562e7bfe806d84747d6518cad714c63ca2e/lua/org-roam/api/node.lua#L370-L381

```lua
                local Capture = require("orgmode.capture")
                local capture = Capture:new({
                    files = files,
                    templates = templates,
                    on_pre_refile = on_pre_refile,
                    on_post_refile = on_post_refile,
                    on_cancel_refile = function()
                        resolve(nil)
                    end,
                })


                return capture:prompt()
```

**Reply by kristijanhusak** [reply, marked as answer]
There is a simpler way. Try this:

```lua
require('orgmode').capture:prompt()
```

Or to open directly a specific template shortcut:

```lua
require('orgmode').capture:open_template_by_shortcut('t')
```

**Comment by rogtino** [comment]
Thank you all.
